workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentTasks.cs

```diff
@@ -0,0 +1,267 @@
+                    });
+                }
+
+                deploymentList.Sort((item1, item2) => string.Compare(item1.Name, item2.Name, StringComparison.Ordinal));
+
+                return deploymentList;
+            });
+
+            return new WrappedTask<List<DeploymentManager.DeploymentInfo>>
+            {
+                Task = task,
+                CancelSource = source
+            };
+        }
+
+        public static async Task<RedirectedProcessResult> RunDeploymentLauncherHelperAsync(CancellationToken token,
+            string workingDirectory,
+            List<string> args,
+            bool redirectStdout = false)
+        {
+            var outputOptions = OutputRedirectBehaviour.RedirectStdErr;
+            if (redirectStdout)
+            {
+                outputOptions |= OutputRedirectBehaviour.RedirectStdOut;
+            }
+
+            var processResult = await RedirectedProcess.Command(Tools.Common.DotNetBinary)
+                .WithArgs(ConstructArguments(args))
+                .RedirectOutputOptions(outputOptions)
+                .InDirectory(workingDirectory)
+                .WithTimeout(DeploymentLauncherTimeout)
+                .RunAsync(token);
+
+            if (processResult.ExitCode == 0 || token.IsCancellationRequested)
+            {
+                return processResult;
+            }
+
+            // Examine the failure reason.
+            var failureReason = processResult.Stdout.Count > 0 ? processResult.Stdout[0] : "";
+            if (failureReason == "<error:unauthenticated>")
+            {
+                // The reason this task failed is because we are authenticated. Try authenticating.
+                Debug.Log(
+                    "Failed to connect to the SpatialOS platform due to being unauthenticated. Running `spatial auth login` then retrying the last operation...");
+                var spatialAuthLoginResult = await RedirectedProcess.Command(Tools.Common.SpatialBinary)
+                    .WithArgs(new string[] { "auth", "login", "--json_output" })
+                    .RedirectOutputOptions(OutputRedirectBehaviour.RedirectStdErr |
+                        OutputRedirectBehaviour.ProcessSpatialOutput)
+                    .InDirectory(workingDirectory)
+                    .RunAsync(token);
+
+                if (spatialAuthLoginResult.ExitCode == 0)
+                {
+                    // Re-run the task.
+                    processResult = await RedirectedProcess.Command(Tools.Common.DotNetBinary)
+                        .WithArgs(ConstructArguments(args))
+                        .RedirectOutputOptions(OutputRedirectBehaviour.RedirectStdErr)
+                        .InDirectory(workingDirectory)
+                        .WithTimeout(DeploymentLauncherTimeout)
+                        .RunAsync(token);
+                }
+                else
+                {
+                    Debug.Log("Failed to run `spatial auth login`.");
```

Comment:
can we log more information on why it failed?
and it probably should be a `LogError`

Reply:
We can't print the toolbelt log I suppose